urllib3/exceptions.py

```diff
@@ -180,6 +180,14 @@ class SNIMissingWarning(HTTPWarning):
     pass
 
 
+class DependencyWarning(HTTPWarning):
```

Comment:
Unless we do:

```py
class Urllib3Warning(Warning):
    ...

HTTPWarning = Urllib3Warning
```

It would be backwards compatible but I don't like that very much anyway because HTTPWarnings still catch dependency warnings. 

Bleh